packages/ra-core/src/routing/AdminRouter.tsx

```diff
@@ -35,6 +36,10 @@ const DummyRouter = ({
     basename?: string;
 }) => <>{children}</>;
 
+const routerProviderFuture: Partial<Pick<FutureConfig, 'v7_startTransition'>> =
+    // @ts-expect-error react-router types are not up-to-date
+    { v7_startTransition: false, v7_relativeSplatPath: false };
```

Comment:
react-router types **are** up-to-date, but you forgot to add `v7_relativeSplatPath` in your `Pick`